src/QmlControls/MultiVehicleFactValueGrid.qml

```diff
@@ -0,0 +1,201 @@
+
+                        Repeater {
+                            id:     labelRepeater
+                            model:  object
+
+                            InstrumentValueLabel {
+                                Layout.fillHeight:      true
+                                Layout.alignment:       Qt.AlignRight
+                                instrumentValueData:    object
+                            }
+                        }
+
+                        Repeater {
+                            id:     valueRepeater
+                            model:  object
+
+                            property real   _index:     index
+                            property real   maxWidth:   0
+                            property var    lastCheck:  new Date().getTime()
+
+                            function recalcWidth() {
+                                var newMaxWidth = 0
+                                for (var i=0; i<valueRepeater.count; i++) {
+                                    newMaxWidth = Math.max(newMaxWidth, valueRepeater.itemAt(i).contentWidth)
+                                }
+                                maxWidth = Math.min(maxWidth, newMaxWidth)
+                            }
+
+                            InstrumentValueValue {
+                                Layout.fillHeight:      true
+                                Layout.alignment:       Qt.AlignLeft
+                                Layout.preferredWidth:  valueRepeater.maxWidth
+                                instrumentValueData:    object
+
+                                property real lastContentWidth
+
+                                Component.onCompleted:  {
+                                    valueRepeater.maxWidth = Math.max(valueRepeater.maxWidth, contentWidth)
+                                    lastContentWidth = contentWidth
+                                }
+
+                                onContentWidthChanged: {
+                                    valueRepeater.maxWidth = Math.max(valueRepeater.maxWidth, contentWidth)
+                                    lastContentWidth = contentWidth
+                                    var currentTime = new Date().getTime()
+                                    if (currentTime - valueRepeater.lastCheck > 30 * 1000) {
+                                        valueRepeater.lastCheck = currentTime
+                                        valueRepeater.recalcWidth()
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            ColumnLayout {
+                spacing: 1
+                visible: settingsUnlocked
+
+                QGCButton {
+                    Layout.preferredWidth:  ScreenTools.minTouchPixels
+                    Layout.fillHeight:      true
+                    topPadding:             0                
```

Comment:
I'm assuming `padding: 0` doesn't work because QGCButton farts around with the individual paddings?

Reply:
this entire file was copied, so I copied the way it was done there. This comment is no longer relevant because I got rid of MulitVehicleFactValueGrid entirely (found a way to reuse the existing HorizontalFactValueGrid)